src/clj/fluree/db/query/exec/select.cljc

```diff
@@ -268,3 +283,17 @@
   (or (instance? AggregateSelector selector)
       (and (instance? AsSelector selector)
            (:aggregate? selector))))
+
+(defn wrap-construct
+  [{:keys [orig-context context]} results]
+  (let [id-key (json-ld/compact const/iri-id context)]
+    (cond-> {"@graph" (->> results
+                           (sort-by #(get % id-key))
+                           (partition-by #(get % id-key))
+                           (mapv display/nest-multicardinal-values))}
```

Comment:
This sorting, partitioning, and nesting is not strictly required by the spec, but may greatly improve usability. If users are mainly using `construct` to directly inject the results into another graph processor, then this is probably unnecessary. If they are trying to use the json-ld document directly, this is probably very useful. Until we discover which use case is more common I've tried to strike a middle path.